lib/debug/server_dap.rb

```diff
@@ -702,9 +702,10 @@ def process_dap args
         frames = []
         @target_frames.each_with_index do |frame, i|
           next if i < start_frame
-          break if (levels -= 1) < 0
 
           path = frame.realpath || frame.path
+          next if skip_path?(path)
+          break if (levels -= 1) < 0
```

Comment:
This now needs to be after deciding to skip or not, or we'd return less frames that required